grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -644,9 +645,18 @@ private ConsistencyLevel getTracingConsistency(BatchParameters parameters) {
           ResponseAndTraceId responseAndTraceId = new ResponseAndTraceId();
           Response.Builder responseBuilder = makeResponseBuilder(result);
           handleTraceId(result.getTracingId(), parameters, responseAndTraceId);
-          if (result.kind != Kind.Void) {
+
+          if (result.kind != Kind.Void && result.kind != Kind.Rows) {
             throw Status.INTERNAL.withDescription("Unhandled result kind").asException();
           }
+
+          if (result.kind == Kind.Rows) {
+            Payload.Type type = validateAndGetTypeForBatchQueries(batch);
```

Comment:
Maybe type validation could be done earlier (before the queries are executed)?

Reply:
done